plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaTable.java

```diff
@@ -0,0 +1,213 @@
+                    .build());
+        }
+
+        public Builder addColumn(String name, Object type, boolean nullable, Optional<String> comment, Map<String, Object> metadata)
+        {
+            columns.add(new Column(name, type, nullable, comment, metadata, Optional.empty()));
+            return this;
+        }
+
+        public Builder renameColumn(String source, String target)
+        {
+            checkArgument(columns.stream().noneMatch(column -> column.name.equalsIgnoreCase(target)), "Column already exists: %s", target);
+
+            Column column = findColumn(source);
+            int index = columns.indexOf(column);
+            verify(index >= 0, "Unexpected column index");
+
+            Column newColumn = new Column(target, column.type, column.nullable, column.comment, column.metadata, column.generation);
+            columns.set(index, newColumn);
+            return this;
+        }
+
+        public Builder removeColumn(String name)
+        {
+            Column column = findColumn(name);
+            boolean removed = columns.remove(column);
+            checkState(removed, "Failed to remove '%s' from %s", name, columns);
+            return this;
+        }
+
+        public Builder setColumnComment(String name, Optional<String> comment)
+        {
+            requireNonNull(comment, "comment is null");
+            Column oldColumn = findColumn(name);
+            Column newColumn = new Column(oldColumn.name, oldColumn.type, oldColumn.nullable, comment, oldColumn.metadata, oldColumn.generation);
+            columns.set(columns.indexOf(oldColumn), newColumn);
+            return this;
+        }
+
+        public Builder dropNotNullConstraint(String name)
+        {
+            Column oldColumn = findColumn(name);
+            verify(!oldColumn.nullable, "Column '%s' is already nullable", name);
+            Column newColumn = new Column(oldColumn.name, oldColumn.type, true, oldColumn.comment, oldColumn.metadata, oldColumn.generation);
+            columns.set(columns.indexOf(oldColumn), newColumn);
+            return this;
+        }
+
+        private Column findColumn(String name)
+        {
+            requireNonNull(name, "name is null");
+            return columns.stream().filter(column -> column.name.equals(name)).collect(onlyElement());
+        }
+
+        public DeltaTable build()
+        {
+            return new DeltaTable(columns, constraints);
+        }
+    }
+
+    @SuppressWarnings("UnusedVariable")
+    private record Column(String name, Object type, boolean nullable, Optional<String> comment, @Nullable Map<String, Object> metadata, Optional<String> generation)
+    {
+        private Column
+        {
+            checkArgument(!name.isEmpty(), "name is empty");
+            requireNonNull(type, "type is null");
+            requireNonNull(comment, "comment is null");
+            requireNonNull(generation, "generation is null");
```

Comment:
generation -> generationExpression

Reply:
Downstream uses `generation` as the variable name.